core/src/main/java/edu/wpi/grip/core/operations/composite/ContoursReport.java

```diff
@@ -1,18 +1,31 @@
 package edu.wpi.grip.core.operations.composite;
 
-import static org.bytedeco.javacpp.opencv_core.*;
+import edu.wpi.grip.core.operations.networktables.NTPublishable;
+import edu.wpi.grip.core.operations.networktables.NTValue;
+
+import java.util.Optional;
+
+import static org.bytedeco.javacpp.opencv_core.MatVector;
+import static org.bytedeco.javacpp.opencv_core.Rect;
+import static org.bytedeco.javacpp.opencv_imgproc.boundingRect;
+import static org.bytedeco.javacpp.opencv_imgproc.contourArea;
 
 /**
  * The output of {@link FindContoursOperation}.  This stores a list of contours (which is basically a list of points) in
  * OpenCV objects, as well as the width and height of the image that the contours are from, to give context to the
  * points.
  */
-public final class ContoursReport {
-    private int rows, cols;
-    private MatVector contours = new MatVector();
+public final class ContoursReport implements NTPublishable {
+    private final int rows, cols;
+    private final MatVector contours;
+    private Optional<Rect[]> boundingBoxes = Optional.empty();
```

Comment:
`volatile` so that when a thread writes to it then it is guaranteed that another thread will see it.